Relocation of part of the utilities from TrackingTypes.h (issue #30090) [1/3]

PetrilloAtWork opened the pull request:
This is the proposed resolution of [LArSoft issue #30090](https://cdcvs.fnal.gov/redmine/issues/30090).

The utilities depending on `lardataalg` functionality have been relocated from `lardataobj/Utilities/TrackingTypes.h` to the new header `lardataalg/Utilities/TrackingTypeUtils.h`.
One unused utility function has been removed (`convertVecPointToTVec3()`).

Three pull requests are involved:
 * LArSoft/lardataobj#54 (this one) removing the functions with unacceptable dependencies;
 * LArSoft/lardataalg#53 where those functions are moved to;
 * LArSoft/larreco#88 for fixes consequent to the breaking change.

## Conversion guide

This is technically a **breaking change**:

* If your program needs one of the utilities which have not changed location, you don't need to do anything
    * but the removal of `larcorealg::Geometry` dependency might expose an existing `CMakeLists.txt` bug, that may require the addition of `larcorealg::Geometry` in the `LIBRARIES` list of existing libraries as appropriate (or `ROOT::Hist` which was indirectly also provided). 
* If your program needs one of the relocated libraries:
    1. inculsion of the header `lardataalg/Utilities/TrackingTypeUtils.h` needs to be added (rarely, it can legitimately _replace_ `TrackingTypes.h` one);
    2. the library `lardataalg::UtilitiesHeaders` needs to be added to the library link list (`LIBRARIES`, choosing `PUBLIC`/`PRIVATE`/`INTERFACE` according to the usual rules).
* If your code needs `recob::tracking::convertVecPointToTVec3()`, report that need to LArSoft team since that function was deemed unused and therefore removed.

Pull requests have been opened in all LArSoft repositories where GitHub reports the usage of one of the functions (except `toPoint()`/`toVector()` which haven't been followed because of too many false positives).
Experiment code was not surveyed.

PetrilloAtWork committed:
Removed dependency on lardataalg from lardataobj

Part of TrackingTypes.h was moved to lardataalg.

This is solution to LArSoft issue #30090.

PetrilloAtWork committed:
The curious result of format-code

FNALbuild commented:
A new Pull Request was created by  @PetrilloAtWork (Gianluca Petrillo) for develop.

It involves the following packages:

lardataobj


@LArSoft/level-1-managers, @LArSoft/level-2-managers can you please review it and eventually sign? Thanks.

cms-bot commands are listed <a href="https://cdcvs.fnal.gov/redmine/projects/larsoft/wiki/Pull_request_comments_that_trigger_CI_actions">here</a>

FNALbuild commented:
The code-checks are being triggered in jenkins.

FNALbuild commented:
+code-checks

lgarren commented:
trigger build with pull requests https://github.com/LArSoft/lardataobj/pull/54, https://github.com/LArSoft/lardataalg/pull/53, https://github.com/LArSoft/larreco/pull/88

FNALbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) LArSoft/lardataalg#53,LArSoft/larreco#88

FNALbuild commented:
+LArSoft tests OK on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:LArSoft/view_builds/index?offset=0&builds=lar_ci/28178&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=&builds=

FNALbuild commented:
+LArSoft tests OK on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:LArSoft/view_builds/index?offset=0&builds=lar_ci/28178&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=&builds=

FNALbuild commented:
+LArSoft tests OK on slf7 for c14:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:LArSoft/view_builds/index?offset=0&builds=lar_ci/28179&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=&builds=

FNALbuild commented:
-SBND tests failed on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/18892&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/28178&builds=

FNALbuild commented:
-ICARUS tests failed on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/13335&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/28178&builds=

FNALbuild commented:
-ICARUS tests failed on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/13335&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/28178&builds=

FNALbuild commented:
-uBooNE tests failed on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:uBooNE/view_builds/index?offset=0&builds=uboone_ci/12970&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/28178&builds=

FNALbuild commented:
-uBooNE tests failed on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:uBooNE/view_builds/index?offset=0&builds=uboone_ci/12970&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/28178&builds=

FNALbuild commented:
-DUNE tests failed on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:DUNE/view_builds/index?offset=0&builds=dune_ci/20296&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/28178&builds=

FNALbuild commented:
-DUNE tests failed on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:DUNE/view_builds/index?offset=0&builds=dune_ci/20296&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/28178&builds=

lgarren commented:
@PetrilloAtWork  This is a breaking change.  Please provide PRs for the experiments to accommodate the changes introduced by these PRs.

PetrilloAtWork commented:
I have downloaded, configured and built ICARUS code, `sbn_suite`, `dune_suite` and `uboone_suite`, and set pull requests accordingly. ArgoNeuT and LArIAT code appears not to be up to date, so I assume they may be forgone.
This is a very high barrier to people volunteering to fix LArSoft issues.

lgarren commented:
@PetrilloAtWork many thanks.  We have always had this requirement for breaking changes.  Would you please list all the PRs so we can run the CI and also communicate that list officially to the collaboration.

PetrilloAtWork commented:
The list:
* Updates for lardataobj PR #54 DUNE/duneprototypes#97
* Updates for lardataobj PR #54 DUNE/dunereco#167
* Updates for lardataobj PR #54 DUNE/dunesim#101
* Updates for lardataobj PR #54 DUNE/protoduneana#66
* Updates for lardataobj PR #54 uboone/ubana#74
* Updates for lardataobj PR #54 uboone/ublite#3
* Added missing ROOT library (exposed by LArSoft/lardataobj PR #54) uboone/ubreco#34
* Added missing LArSoft library (exposed by LArSoft/lardataobj PR #54) uboone/ubevt#4
* Added missing ROOT library (exposed by LArSoft/lardataobj PR #54) uboone/ubcore#4

SBN one was an existing missing dependency that was unveiled by this change, and it has already been merged.

lgarren commented:
Thanks @PetrilloAtWork  Apologies for not realizing you had already posted it.

lgarren commented:
triggger build with pull requests https://github.com/LArSoft/lardataobj/pull/54, https://github.com/LArSoft/lardataalg/pull/53, https://github.com/LArSoft/larreco/pull/88, https://github.com/DUNE/duneprototypes/pull/97, https://github.com/DUNE/dunereco/pull/167, https://github.com/DUNE/dunesim/pull/101, https://github.com/DUNE/protoduneana/pull/66, https://github.com/uboone/ubana/pull/74, https://github.com/uboone/ublite/pull/3, https://github.com/uboone/ubreco/pull/34, https://github.com/uboone/ubevt/pull/4, https://github.com/uboone/ubcore/pull/4, https://github.com/SBNSoftware/sbndcode/pull/880

lgarren commented:
triggger build with pull requests https://github.com/LArSoft/lardataobj/pull/54, https://github.com/LArSoft/lardataalg/pull/53, https://github.com/LArSoft/larreco/pull/88, https://github.com/DUNE/duneprototypes/pull/97, https://github.com/DUNE/dunereco/pull/167, https://github.com/DUNE/dunesim/pull/101, https://github.com/DUNE/protoduneana/pull/66, https://github.com/uboone/ubana/pull/74, https://github.com/uboone/ublite/pull/3, https://github.com/uboone/ubreco/pull/34, https://github.com/uboone/ubevt/pull/4, https://github.com/uboone/ubcore/pull/4, https://github.com/SBNSoftware/sbndcode/pull/880

lgarren commented:
trigger build with pull requests https://github.com/LArSoft/lardataobj/pull/54, https://github.com/LArSoft/lardataalg/pull/53, https://github.com/LArSoft/larreco/pull/88, https://github.com/DUNE/duneprototypes/pull/97, https://github.com/DUNE/dunereco/pull/167, https://github.com/DUNE/dunesim/pull/101, https://github.com/DUNE/protoduneana/pull/66, https://github.com/uboone/ubana/pull/74, https://github.com/uboone/ublite/pull/3, https://github.com/uboone/ubreco/pull/34, https://github.com/uboone/ubevt/pull/4, https://github.com/uboone/ubcore/pull/4, https://github.com/SBNSoftware/sbndcode/pull/880

FNALbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) LArSoft/lardataalg#53,LArSoft/larreco#88,DUNE/duneprototypes#97,DUNE/dunereco#167,DUNE/dunesim#101,DUNE/protoduneana#66,uboone/ubana#74,uboone/ublite#3,uboone/ubreco#34,uboone/ubevt#4,uboone/ubcore#4,SBNSoftware/sbndcode#880

FNALbuild commented:
+LArSoft tests OK on slf7 for c14:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:LArSoft/view_builds/index?offset=0&builds=lar_ci/28606&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=&builds=

FNALbuild commented:
+LArSoft tests OK on slf7 for c14:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:LArSoft/view_builds/index?offset=0&builds=lar_ci/28606&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=&builds=

FNALbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) LArSoft/lardataalg#53,LArSoft/larreco#88,DUNE/duneprototypes#97,DUNE/dunereco#167,DUNE/dunesim#101,DUNE/protoduneana#66,uboone/ubana#74,uboone/ublite#3,uboone/ubreco#34,uboone/ubevt#4,uboone/ubcore#4,SBNSoftware/sbndcode#880

FNALbuild commented:
+LArSoft tests OK on slf7 for c14:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:LArSoft/view_builds/index?offset=0&builds=lar_ci/28608&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=&builds=

FNALbuild commented:
+LArSoft tests OK on slf7 for c14:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:LArSoft/view_builds/index?offset=0&builds=lar_ci/28608&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=&builds=

FNALbuild commented:
+LArSoft tests OK on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:LArSoft/view_builds/index?offset=0&builds=lar_ci/28607&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=&builds=

FNALbuild commented:
+LArSoft tests OK on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:LArSoft/view_builds/index?offset=0&builds=lar_ci/28607&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=&builds=

FNALbuild commented:
-uBooNE tests failed on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:uBooNE/view_builds/index?offset=0&builds=uboone_ci/13086&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/28607&builds=

FNALbuild commented:
-uBooNE tests failed on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:uBooNE/view_builds/index?offset=0&builds=uboone_ci/13086&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/28607&builds=

FNALbuild commented:
-SBND tests warning, with build warning,, with ignored warning for build, on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/19349&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/28607&builds=

FNALbuild commented:
-SBND tests warning, with build warning,, with ignored warning for build, on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/19349&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/28607&builds=

FNALbuild commented:
-ICARUS tests failed, with build warning,, with ignored warning for build, on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/13588&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/28607&builds=

FNALbuild commented:
-ICARUS tests failed, with build warning,, with ignored warning for build, on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:ICARUS/view_builds/index?offset=0&builds=icarus_ci/13588&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/28607&builds=

FNALbuild commented:
-DUNE tests failed on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:DUNE/view_builds/index?offset=0&builds=dune_ci/20412&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/28607&builds=

FNALbuild commented:
-DUNE tests failed on slf7 for e26:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:DUNE/view_builds/index?offset=0&builds=dune_ci/20412&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/28607&builds=
